### PR TITLE
Add docs for the configuration input detection opt-out flags for Gradle 8.3

### PR DESCRIPTION
**Pull request by h0tk3y**
This replicates https://github.com/gradle/gradle/pull/25696 with the adjustments for the 8.3 release.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 1]
```diff
@@ -567,6 +567,45 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+** Report issues caused by third-party plugins to the plugin maintainers, and update the plugins once they get fixed.
+
+* For some kinds of configuration inputs, it is possible to use the opt-out options that make Gradle fall back to the earlier behavior, omitting the inputs from detection.
+This temporary workaround is aimed to mitigate performance issues coming from out-of-date plugins.
+
+It is currently possible to temporarily opt out of configuration input detection in the following case:
+
+* Since Gradle 8.1, using many APIs related to the file system is correctly tracked as configuration inputs, including the file system checks, such as `File.exists()` or `File.isFile()`.
++
+For the input tracking to ignore these file system checks on the specific paths, one can provide a Gradle property `org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks` with the list of the paths, relative to the root project directory and separated by `;`.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:591]:
```suggestion
For the input tracking to ignore these file system checks on the specific paths, the Gradle property `org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks`, with the list of the paths, relative to the root project directory and separated by `;`, can be provided.
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 2]
```diff
@@ -567,6 +567,45 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+To ignore multiple paths, use `*` to match arbitrary strings within one segment, or `pass:[**]` across segments.
+Paths starting with `~/` will be considered to be based on the user home directory.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:593]:
```suggestion
Paths starting with `~/` are based on the user home directory.
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 3]
```diff
@@ -567,6 +567,45 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+[source,properties]
+.gradle.properties
+----
+org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks=\
+    ~/.third-party-plugin/*.lock;\
+    ../../externalOutputDirectory/**;\
+    build/analytics.json
+----
+
+One should ignore configuration inputs sparingly, and only if those do not affect the tasks produced by the configuration logic.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:605]:
```suggestion
Ignore configuration inputs sparingly, and only if they do not affect the tasks produced by the configuration logic.
```

**Review by lkasso — APPROVED** [review]
Very minor changes requested for grammatical purposes.

**Comment by h0tk3y** [comment]
@lkasso Thanks! As these changes are mostly a cherry-pick of https://github.com/gradle/gradle/pull/25696 with an unrelated paragraph removed, does it make sense to also apply your suggestions to `master`?

**Comment by h0tk3y** [comment]
@bot-gradle test and merge

**Comment by mlopatkin** [comment]
>As these changes are mostly a cherry-pick of #25696 with an unrelated paragraph removed, does it make sense to also apply your suggestions to `master`?

I'll do that as part of the release -> master merge today.

**Pull request merged**